src/pages/tests/HomePage.test.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { Provider } from 'react-redux';
+import { render } from '@testing-library/react';
+import { configureStore } from '@reduxjs/toolkit';
+import { MemoryRouter } from 'react-router-dom';
+import { expect } from '@jest/globals';
+import HomePage from '../HomePage';
+import dataReducer from '../../redux/data/dataSlice';
+
+describe('HomePage component', () => {
+  test('should render properly', () => {
+    const mockData = [
+      {
+        name: 'Country 1', continents: ['Africa'], population: 1000000, area: 100,
+      },
+      {
+        name: 'Country 2', continents: ['North America'], population: 2000000, area: 200,
+      },
+      {
```

Comment:
- [ ] Good work on testing your application but only the `HomePage` component is tested. Kindly test all your components and also the pure functions in your Redux code. This will ensure that the components and functions are working correctly and producing the expected output

Reply:
Yes Sir. Added the tests accordingly

Reply:
thanks